packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -431,6 +431,7 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
           onStateChange={this.handleStateChange}
           labelId={this.props['aria-labelledby']}
           environment={this.context.target.defaultView}
+          inputId={this.props.searchInput ? this.props.searchInput['id'] : undefined}
```

Comment:
```tsx
<Dropdown searchInput={render => render('boo')} />
<Dropdown searchInput='Oops' />
```

This thing is pretty fragile. Do we have other options?

Reply:
you are right, I need to add another check. this will only work if `searchInput` is an object.